dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/ObjectBundle.java

```diff
@@ -334,24 +334,13 @@ private void addObject( IdentifiableObject object )
 
         Class<? extends IdentifiableObject> realClass = HibernateProxyUtils.getRealClass( object );
 
-        if ( !objects.get( Boolean.TRUE ).containsKey( realClass ) )
-        {
-            objects.get( Boolean.TRUE ).put( realClass, new ArrayList<>() );
-        }
-
-        if ( !objects.get( Boolean.FALSE ).containsKey( realClass ) )
-        {
-            objects.get( Boolean.FALSE ).put( realClass, new ArrayList<>() );
-        }
```

Comment:
Note the subtle change in behaviour here - while the old way would both initialise an persisted and non-persisted lists for the object type even if just one of them is then needed/used the new code only initialised the one actually used. I'd hope if anything this should make the behaviour "more correct".